src/App.tsx

```diff
@@ -1,16 +1,39 @@
 import { useState } from 'react';
 import { Layout } from 'shared';
+import { Routes, Route, Navigate } from 'react-router-dom';
+import { Blog, BlogDetails, Home, Info, Login } from 'views';
+import { BlogPosts } from 'modules';
 
 export const App: React.FC = () => {
   const [isLoggedIn, setIsLogggedIn] = useState<boolean>(false);
 
   function onLogin() {
     setIsLogggedIn(!isLoggedIn);
   }
-
   return (
-    <Layout onLogin={() => onLogin()}>
-      <p>TODO: Add react router</p>
-    </Layout>
+    <Routes>
+      <Route path="/" element={<Layout onLogin={() => onLogin()} />}>
+        <Route index element={<Home />} />
+        <Route path="info" element={<Info />} />
+        <Route
+          path="blog"
+          element={isLoggedIn ? <Blog /> : <Navigate to="/login" />}
+        >
+          <Route
+            index
+            element={isLoggedIn ? <BlogPosts /> : <Navigate to="/login" />}
+          />
+          <Route
+            path=":id"
+            element={isLoggedIn ? <BlogDetails /> : <Navigate to="/login" />}
+          />
+        </Route>
+        <Route
+          path="login"
+          element={!isLoggedIn ? <Login /> : <Navigate to="/blog" />}
+        />
+        <Route path="*" element={<p>There's nothing here!</p>} />
+      </Route>
+    </Routes>
```

Comment:
Ove provjere _isLoggedIn_  unutar blog rute su ti viška, njih si mogla izostaviti i radilo bi ti isto kao i što sada radi. Također, pošto u Blog komponenti renderas samo Outlet po meni se ta Blog komponenta mogla i maknuti (okej bi mi bilo da uz outlet imas jos nesto sto je zajednicko BlogPost i BlogDetails komponenti ) npr:

```
export const Blog: React.FC = () => {
  return (
    <>
      <h1>This is blog post stuff</h1>
      <Outlet />
    </>
  );
};
```